test/Microsoft.Azure.SignalR.Management.Tests/Serialization/AddNewtonsoftFacts.cs

```diff
@@ -0,0 +1,203 @@
+                    .CreateAsync("hubName", default) as ServiceHubContextImpl;
+                await serviceHubContext.Clients.All.SendAsync(methodName, message);
+
+                var sentMessage = connectionFactory.CreatedConnections.Single().Value.SelectMany(conn => ((TestServiceConnection)conn).ReceivedMessages).Single() as BroadcastDataMessage;
+                var protocol = new NewtonsoftJsonHubProtocol(Options.Create(new NewtonsoftJsonHubProtocolOptions() { PayloadSerializerSettings = jsonSerializerSettings }));
+                var exptectedPayload = protocol.GetMessageBytes(new InvocationMessage(methodName, new object[] { message }));
+                var actualPayload = sentMessage.Payloads[protocol.Name];
+                logger.LogInformation(JsonConvert.SerializeObject(jsonSerializerSettings));
+                logger.LogInformation("Expected:  {expected}", Encoding.UTF8.GetString(exptectedPayload.Span));
+                logger.LogInformation("Actual:    {actual}", Encoding.UTF8.GetString(actualPayload.Span));
+                logger.LogInformation("Version:   {version}",Environment.Version.ToString());
+
+                var hubProtocol = serviceHubContext.ServiceProvider.GetRequiredService<IHubProtocol>();
+                logger.LogInformation("Current HubProtocol:  {hubProtocol}",hubProtocol.GetType().FullName);
+                var services = serviceHubContext.ServiceProvider.GetRequiredService<IReadOnlyCollection<ServiceDescriptor>>();
+                foreach(var service in services.Where(s => s.ServiceType == typeof(IHubProtocol)))
+                {
+                    logger.LogInformation("Added HubProtocol:{impl}", service.ImplementationType);
+                }
+
+                Assert.True(exptectedPayload.Span.SequenceEqual(actualPayload.Span));
+
+                await serviceHubContext.DisposeAsync();
+            }
+        }
+
+        private static ServiceHubContextBuilder CreatePersistentServiceHubContextBuilder(out TestServiceConnectionFactory connectionFactory)
+        {
+            connectionFactory = new TestServiceConnectionFactory();
+            var connectionFactoryAlias = connectionFactory;
+            //skip check if endpoint online
+            var routerMock = new Mock<IEndpointRouter>();
+            routerMock.Setup(r => r.GetEndpointsForBroadcast(It.IsAny<IEnumerable<ServiceEndpoint>>())).Returns((IEnumerable<ServiceEndpoint> es) => es);
+            return new ServiceHubContextBuilder()
+                .WithOptions(o =>
+                {
+                    o.ConnectionString = FakeEndpointUtils.GetFakeConnectionString(1).Single();
+                    o.ServiceTransportType = ServiceTransportType.Persistent;
+                })
+                .WithRouter(routerMock.Object)
+                .ConfigureServices(services => services.AddSingleton<IServiceConnectionFactory>(connectionFactoryAlias));
+        }
+
+        #endregion Persistent
+
+        public class HubMethodArgument
```

Comment:
we only have `private sealed class` as inner classes

Reply:
`HubMethodArgument` is used in `[MemberData(nameof(SerializdObject_SerializerSettings))]`  and `MemberData` must be public